### ISSUE TITLE
SVG gradients do not render properly

### ISSUE THREAD

**Issue by jwright6323** [mogest/prawn-svg#153]
Gist: https://gist.github.com/jwright6323/666747665d920d6bc294959be24f6822

I am seeing that gradients are rendered based on the image's location within a page rather than a component's position within the SVG (see example in Gist above).

**Comment by mogest** [comment]
I tried to replicate this but can't.  See [_multiple_gradients.pdf](https://github.com/mogest/prawn-svg/files/14074816/_multiple_gradients.pdf) which shows two of the SVG you provided displayed on top of each other.

Given this is quite old I'll close it for now, but if you can replicate it and provide a code example using only prawn and prawn-svg I'd be happy to look into it.

**Comment by mojavelinux** [comment]
I can confirm this is an issue. If you use the following script, you'll see that even though the SVG moves, the gradient is always draw from the same x,y position on the page, so the second and third SVGs are missing the gradient.

```rb
require 'prawn'
require 'prawn-svg'

Prawn::Document.generate 'out.pdf' do
  svg (File.read 'tmp.drawio.svg'), width: 100
  svg (File.read 'tmp.drawio.svg'), width: 100
  svg (File.read 'tmp.drawio.svg'), width: 100
end
```

**Comment by mogest** [comment]
Interesting!  I had `width: 300`, it appears that if it's `width: 321` then it perfectly aligns and the gradients show.  Thanks, this gives me something to go off.

**Comment by mogest** [comment]
I can replicate it ([_multiple_gradients.pdf](https://github.com/mogest/prawn-svg/files/14149541/_multiple_gradients.pdf)) and the first svg is the one that gets the gradient, no matter where on the screen it is.

Looking at the PDF though (upload to https://pdfprettyprint.com if you want a go), all seems fine.  I initially thought it was reusing the gradient definitions, but no, each rect has its own gradient, and they're all at different co-ordinates.

Looking at the code, I can see nothing except the gradients that get cached in a registry between calls to prawn-svg.

Quite confused.  If you spot anything in the PDF, do let me know...

**Comment by mogest** [comment]
@mojavelinux's test case above appears to be fixed on main (and was also working on 0.35.1).  With @tonymarklove's work I think gradients are in a healthy place!  Please reopen this issue if there are still problems.
